lib/Slic3r/Print/GCode.pm

```diff
@@ -453,6 +453,16 @@ sub process_layer {
         $pp->set('layer_z'   => $layer->print_z);
         $gcode .= Slic3r::ConditionalGCode::apply_math($pp->process($self->print->config->layer_gcode) . "\n");
     }
+    if ($self->print->config->pause_gcode && $self->print->config->pause_heights) {
+        foreach my $height (split / /, $self->print->config->pause_heights) {
+            if ($height && $height eq $layer->print_z) {
```

Comment:
Wouldn't it be appropriate to round the effective height (either up or down, down may be the safer option) if the current layer doesn't land on the prescribed height? 

It's something that can occur often with adaptive slicing turned on or if there's some oddball floating point math going on internally. 

Obviously you'd want to extend the tooltip appropriately.

Reply:
I'll have to play with how this interacts with Adaptive Slicing, but my gut says no. I also work with ideamaker, which has this feature and does such rouding ... and the end result is that I have to manually open my gcode after export to make sure it paused where I wanted it to. They have multiple bugs open because of this. (now ... I have to open the gcode anyway because they don't allow me to customize the code inserted, but that's a different story).

Reply:
Since logically this is intended to be a before-layer gcode, using floor() would seem to be safe (as opposed to round(), which could go up or down and definitely could confuse people). 

That is, if layer 5 spans z=0.1 and z=0.3 and z=0.2 was requested, it would be inserted at 0.1. If z=0.299999999 was requested, it'd still be at z=0.1.